client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,257 @@
+		};
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seo_meta_description: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event ) {
+		const productId = get( event, 'target.id', '' ).replace( 'verification_code_', '' );
+
+		// Build verification services object to send back to API
+		let verificationServicesCodes = this.getCurrentVerificationCodes();
+
+		if ( ! verificationServicesCodes.hasOwnProperty( productId ) ) {
+			return;
+		}
+
+		verificationServicesCodes[ productId ] = get( event, 'target.value', '' );
+
+		this.setState( { verification_services_codes: verificationServicesCodes } );
+	},
+
+	getCurrentVerificationCodes() {
+		return {
+			google: this.refs.verification_code_google.props.value,
+			bing: this.refs.verification_code_bing.props.value,
+			pinterest: this.refs.verification_code_pinterest.props.value,
+			yandex: this.refs.verification_code_yandex.props.value
+		};
+	},
+
+	submitSeoForm( event ) {
+		var site = this.props.site;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		this.setState( { submittingForm: true } );
+
+		const updatedOptions = {
+			seo_meta_description: this.state.seo_meta_description,
+			verification_services_codes: this.state.verification_services_codes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
+						break;
+					default:
+						notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+				}
+				this.setState( { submittingForm: false } );
+			} else {
+				notices.success( this.translate( 'Settings saved!' ) );
+				this.markSaved();
+				this.setState( { submittingForm: false } );
+
+				site.fetchSettings();
+			}
+		} );
+	},
+
+	render() {
+		const { site } = this.props;
+		const { submittingForm, seo_meta_description, verification_services_codes } = this.state;
+
+		const blog_public = get( site, 'settings.blog_public', 1 );
+		const isSitePrivate = parseInt( blog_public, 10 ) !== 1;
+
+		const fetchingSettings = get( site, 'fetchingSettings', false );
+		const isDisabled = isSitePrivate || fetchingSettings || submittingForm;
+		const isMetaError = seo_meta_description && seo_meta_description.length > 160;
+		const sitemapUrl = `https://${ site.slug }/sitemap.xml`;
+		const generalSettingsUrl = `/settings/general/${ site.slug }`;
+
+		// The API returns 'false' for an empty array value, so we force it to an empty string if needed
+		const googleCode = get( verification_services_codes, 'google' ) || '';
+		const bingCode = get( verification_services_codes, 'bing' ) || '';
+		const pinterestCode = get( verification_services_codes, 'pinterest' ) || '';
+		const yandexCode = get( verification_services_codes, 'yandex' ) || '';
```

Comment:
@rodrigoi I'm more reluctant to destructure that many nested levels, just because of general code confusion. When I _do_ do this, I like to split it up into separate lines.

In any case, as per my comment response to @gwwar, defaulting won't work here because the API is returning `false` and not `undefined`